lib/shippo/api/api_object.rb

```diff
@@ -78,7 +74,7 @@ def mk_opts(property)
 
       # list of allowed properties, of a given type.
       PROPS_ID         = %i(id).freeze
-      PROPS_CATEG      = %i(state purpose source status results).freeze
+      PROPS_CATEG      = %i(state status results).freeze
```

Comment:
per Guozhen's notes: object_purpose and object_source are removed everywhere, object_state is removed from address, shipment and rates